app/customer.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass
+import math
+
+from app.car import Car
+from app.shop import Shop
+
+
+@dataclass
+class Customer:
+    name: str
+    product_cart: dict
+    location: list[int]
+    money: int | float
+    car: Car
+
+    def have_money(self) -> str:
+        return f"{self.name} has {self.money} dollars"
+
+    def trip_cost(self, shop: Shop, fuel_price: float) -> int | float:
+        distance = math.sqrt(
+            (self.location[0] - shop.location[0]) ** 2
+            + (self.location[1] - shop.location[1]) ** 2)
+        fuel = distance / 100 * self.car.fuel_consumption
+        return round(fuel * fuel_price * 2, 2)
+
+    def product_cost(self, shop: Shop) -> int | float:
+        if not shop.products:

```

Comment:
The check `if not shop.products:` assumes that `shop.products` is always a dictionary. Ensure that the `Shop` class initializes `products` as a dictionary to avoid potential errors. This is confirmed in the `Shop` class definition.